From 6abd26adbc5d6f5732d3cad933a2449648e1b5de Mon Sep 17 00:00:00 2001
From: Vinith Kumar R <quic_vinithku@quicinc.com>
Date: Tue, 1 Oct 2024 16:16:48 +0530
Subject: [PATCH 4/4] wifi: ath12k: Add NULL check for skb2 before
 ieee80211_free_txskb call

Currently, in the ieee80211_tx_prepare_skb function, variable skb2
may be NULL after __skb_dequeue function call. So need to do NULL
check for skb2 before ieee80211_free_txskb function call to prevent
NULL pointer dereference

Signed-off-by: Vinith Kumar R <quic_vinithku@quicinc.com>
---
 net/mac80211/tx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index 879e78a..56f5471 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -1982,7 +1982,8 @@ bool ieee80211_tx_prepare_skb(struct ieee80211_hw *hw,
 	/* this function isn't suitable for fragmented data frames */
 	skb2 = __skb_dequeue(&tx.skbs);
 	if (WARN_ON(skb2 != skb || !skb_queue_empty(&tx.skbs))) {
-		ieee80211_free_txskb(hw, skb2);
+		if(skb2)
+			ieee80211_free_txskb(hw, skb2);
 		ieee80211_purge_tx_queue(hw, &tx.skbs);
 		return false;
 	}
-- 
2.34.1

